src/components/Table/Table.tsx

```diff
@@ -0,0 +1,257 @@
+import * as React from 'react';
+import { themr, ThemedComponentClass } from 'react-css-themr';
+import { classNames } from '@shopify/react-utilities/styles';
+
+import { TABLE } from '../ThemeIdentifiers';
+
+import TableHeader from './TableHeader';
+import TableHead from './TableHead';
+import TableBody from './TableBody';
+import TableRow from './TableRow';
+import TableData from './TableData';
+
+import * as baseTheme from './Table.scss';
+
+export interface Props {
+  // To make table bordered
+  bordered?: boolean;
+  // Column config, which renders the header
+  column?: any;
+  // Get the data & use it to populate tds
+  data?: any;
+  // On load by default it will be sorted by this field
+  defaultSortField?: string;
+  // Value could be 'asc' || 'desc'
+  defaultSortOrder?: string;
+  // Highlight rows on hover
+  highlight?: boolean;
+  // Make table responsive
+  responsive?: boolean;
+  // Flag to indentify if table is sortable, if passed "all" then add sorting to all the columns
+  sorting?: boolean | string;
+  // Set greyed background for odd rows
+  striped?: boolean;
+  theme?: any;
+}
+
+export interface State {
+  data?: any;
+  sort?: any;
```

Comment:
@dheeraja00 We need to explicity define an interface for this.  Check this out: https://www.typescriptlang.org/docs/handbook/interfaces.html